### PR TITLE
Validate parameters for vault operator init  

### PR DESCRIPTION
**Pull request by akshya96**
https://hashicorp.atlassian.net/browse/VAULT-951

The "vault operator" command should not operate if there are unused command-line flags, for example:
vault operator init --recovery-threshold=1 --recovery-shares=1
should return an error because auto unseal is not being used (the correct flags are -key-shares and -key threshold.)

### CODE REVIEW

**Comment by akshya96** [comment]
> I agree that you have to remove the defaults from the flag definitions, but I wouldn't push everything into the api as you're doing. People often use different versions of the CLI and server, and this could break backwards compat.
> 
> We might have to make the operator init command query the sys/seal-status endpoint to determine whether shamir is in use or not, so it can make the right decision about which defaults to use when the user doesn't specify any share/threshold flags.

I agree, I did not think of different versions of cli and server. I have changed the cli to set default values based on use of auto unseal instead of setting default values for all fields.

File: http/sys_init.go [hunk 1]
```diff
@@ -128,3 +135,30 @@ type InitResponse struct {
 type InitStatusResponse struct {
 	Initialized bool `json:"initialized"`
 }
+
+// Validates if the right parameters are used based on AutoUnseal
+func validateInitParameters(core *vault.Core, req InitRequest) error {
+	var recoveryFlagsPresent bool
+	var barrierFlagsPresent bool
+
+	if req.SecretShares != 0 || req.SecretThreshold != 0 || req.StoredShares != 0 || len(req.PGPKeys) != 0 {
```
Comment (ncabatoff) [inline http/sys_init.go:144]: I would probably omit StoredShared, I don't think it's meaningful anymore.

File: http/sys_init.go [hunk 2]
```diff
@@ -128,3 +135,30 @@ type InitResponse struct {
...
+		barrierFlagsPresent = true
+	}
+
+	if req.RecoveryShares != 0 || req.RecoveryThreshold != 0 || len(req.RecoveryPGPKeys) != 0 {
+		recoveryFlagsPresent = true
+	}
+
+	switch core.SealAccess().RecoveryKeySupported() {
+	case true:
+		if barrierFlagsPresent {
```
Comment (ncabatoff) [inline http/sys_init.go:154]: We could be a little more helpful/explicit by specifying the actual param name that we don't like.  Failing that, we could say something in the error like "secret_shares, secret_threshold, and pgp_keys are not applicable to %s" where %s is seal type.

**Review by ncabatoff — APPROVED** [review]

**Review by ccapurso — APPROVED** [review]
Looks good!

**Comment by akshya96** [comment]
Added a new unit test for auto unseal

**Review by ccapurso — APPROVED** [review]
Tests look good!

**Pull request merged**